miio/airhumidifier_jsq.py

```diff
@@ -0,0 +1,330 @@
+            )
+
+        return AirHumidifierStatus(
+            {k: v for k, v in zip(properties, values)}, self.device_info
+        )
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.send("set_start", [1])
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.send("set_start", [0])
+
+    @command(
+        click.argument("mode", type=EnumType(OperationMode, False)),
+        default_output=format_output("Setting mode to '{mode.value}'"),
+    )
+    def set_mode(self, mode: OperationMode):
+        """Set mode."""
+        value = mode.value
+        if value not in (
+            e.value
+            for e in [
+                OperationMode.Intelligent,
+                OperationMode.Level1,
+                OperationMode.Level2,
+                OperationMode.Level3,
+                OperationMode.Level4,
+            ]
+        ):
+            raise AirHumidifierException(
+                "{} is not a valid OperationMode value".format(value)
+            )
+
+        return self.send("set_mode", [value])
+
+    @command(
+        click.argument("brightness", type=EnumType(LedBrightness, False)),
+        default_output=format_output("Setting LED brightness to {brightness}"),
+    )
+    def set_led_brightness(self, brightness: LedBrightness):
+        """Set led brightness."""
+        value = brightness.value
+        if value not in (
+            e.value for e in [LedBrightness.Off, LedBrightness.Low, LedBrightness.High]
+        ):
+            raise AirHumidifierException(
+                "{} is not a valid LedBrightness value".format(value)
+            )
+
+        return self.send("set_brightness", [value])
+
+    @command(
+        click.argument("led", type=bool),
+        default_output=format_output(
+            lambda led: "Turning on LED" if led else "Turning off LED"
+        ),
+    )
+    def set_led(self, led: bool):
+        """Turn led on/off."""
+        if led:
+            return self.set_led_brightness(LedBrightness.High)
+        else:
+            return self.set_led_brightness(LedBrightness.Off)
+
+    @command(
+        click.argument("buzzer", type=bool),
+        default_output=format_output(
+            lambda buzzer: "Turning on buzzer" if buzzer else "Turning off buzzer"
+        ),
+    )
+    def set_buzzer(self, buzzer: bool):
+        """Set buzzer on/off."""
+        if buzzer:
+            return self.send("set_buzzer", [1])
+        else:
+            return self.send("set_buzzer", [0])
+
+    @command(
+        click.argument("lock", type=bool),
+        default_output=format_output(
+            lambda lock: "Turning on child lock" if lock else "Turning off child lock"
+        ),
+    )
+    def set_child_lock(self, lock: bool):
+        """Set child lock on/off."""
+        if lock:
+            return self.send("set_lock", [1])
```

Comment:
You could use only a single `send` here with `[int(lock)]` to cover both cases.

Reply:
Ok, I agree that cleanup is required here. 

Instead of `int(lock)` I'd rather use smth more type safe. Your version assumes that `lock` is `bool` or `number`, but smb could call this API with wrong type, e.g. `lock="on"` and we will get an exception `ValueError: invalid literal for int() with base 10: 'on'`. I suggest to use smth more type-safe here, e.g. `[int(bool(lock))]` or `[1 if lock else 0]`

Reply:
Fixed, see updated version.

Reply:
I think click does automatic conversion for bool-like types, but that's fine too :-)